expression/builtin_math.go

```diff
@@ -172,34 +172,26 @@ type floorFunctionClass struct {
 }
 
 func (c *floorFunctionClass) getFunction(args []Expression, ctx context.Context) (builtinFunc, error) {
-	sig := &builtinFloorSig{newBaseBuiltinFunc(args, ctx)}
+	bf, err := newBaseBuiltinFuncWithTp(args, ctx, tpReal, tpReal)
```

Comment:
Hi @XuHuaiyu, thank you for pointing it out.
I have checked and found that if the arg is int or decimal type, the `floor` function will return int type data to client side. For other cases, the `floor` function will return double. I have update my branch. Please take a look. Thanks :)